crates/api_common/src/post.rs

```diff
@@ -16,6 +16,17 @@ use serde_with::skip_serializing_none;
 #[cfg(feature = "full")]
 use ts_rs::TS;
 
+#[skip_serializing_none]
+#[derive(Debug, Serialize, Deserialize, Clone, Default)]
+#[cfg_attr(feature = "full", derive(TS))]
+#[cfg_attr(feature = "full", ts(export))]
+/// Update tags for a post.
```

Comment:
Would probably be good to add a comment about how either mods or post creators can do this.